app/Rest/Employees_Controller.php

```diff
@@ -0,0 +1,475 @@
+
+		$post_id = wp_insert_post( wp_slash( (array) $prepared_post ), true );
+
+		if ( is_wp_error( $post_id ) ) {
+			$post_id->add_data( array( 'status' => 400 ) );
+
+			if ( 'db_insert_error' === $post_id->get_error_code() ) {
+				$post_id->add_data( array( 'status' => 500 ) );
+			}
+
+			return $post_id;
+		}
+
+		$schema = $this->get_item_schema();
+
+		if ( ! empty( $schema['properties']['acf_field'] ) && isset( $request['acf_field'] ) ) {
+			$acf_update = $this->update_acf_value( $request['acf_field'], $post_id );
+
+			if ( is_wp_error( $acf_update ) ) {
+				return $acf_update;
+			}
+		}
+
+		$request->set_param( 'context', 'edit' );
+
+		return $this->response(
+			[
+				'status'      => 'Employee has been created',
+				'id'          => $post_id,
+				'title'       => ! empty( $prepared_post->post_title ) ? $prepared_post->post_title : '',
+				'post_status' => $prepared_post->post_status,
+			],
+			201,
+			[
+				'Location' => rest_url( sprintf( '%s/%s/%d', $this->namespace, $this->resource_name, $post_id ) ),
+			]
+		);
+	}
+
+	/**
+	 * Checks if a given request has access to create a post.
+	 *
+	 * @param \WP_REST_Request $request Full details about the request.
+	 *
+	 * @return true|\WP_Error True if the request has access to create items, WP_Error object otherwise.
+	 */
+	public function create_item_permissions_check( $request ) {
+		if ( ! empty( $request['id'] ) ) {
+			return new \WP_Error( 'rest_post_exists', __( 'Cannot create existing post.' ), array( 'status' => 400 ) );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Get_item
+	 *
+	 * @param \WP_REST_Request $request Full details about the request.
+	 *
+	 * @return \WP_Error|\WP_REST_Response
+	 */
+	public function get_item( $request ) {
+		$post = $this->get_employee( $request['id'] );
+		if ( is_wp_error( $post ) ) {
+			return $post;
+		}
+
+		return $this->response( $post );
+	}
+
+	/**
+	 * Get_item
+	 *
+	 * @param \WP_REST_Request $request Full details about the request.
+	 *
+	 * @return \WP_Error|\WP_REST_Response
+	 */
+	public function update_item( $request ) {
+		$prepared_post = $this->prepare_item_for_database( $request );
+
+		if ( is_wp_error( $prepared_post ) ) {
+			return $prepared_post;
+		}
+
+		$prepared_post->post_type   = Employee_Post_Type::$ID;
+		$prepared_post->post_status = Employee_Post_Type::STATUS_PUBLISH;
+
+		$post_id = wp_update_post( wp_slash( (array) $prepared_post ), true );
+
+		if ( is_wp_error( $post_id ) ) {
+			$post_id->add_data( array( 'status' => 400 ) );
+
+			if ( 'db_insert_error' === $post_id->get_error_code() ) {
+				$post_id->add_data( array( 'status' => 500 ) );
+			}
+
+			return $post_id;
+		}
+
+		$schema = $this->get_item_schema();
+
+		if ( ! empty( $schema['properties']['acf_field'] ) && isset( $request['acf_field'] ) ) {
+			$acf_update = $this->update_acf_value( $request['acf_field'], $post_id );
+
+			if ( is_wp_error( $acf_update ) ) {
+				return $acf_update;
+			}
+		}
```

Comment:
you have similar code for insert/update after wp_insert_post()/wp_update_post() - move this part to a separate method, like update_postmeta() or something and method to prepare response. so you will get something like that:

$post_id = wp_update_post(...);

if ( ! is_wp_error($post_id) ) { $this->update_postmeta(...) }

return $this->editable_response($post_id, ["response in case post_id is valid and not is error"])